plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -660,6 +688,174 @@ private static boolean isPassthroughQuery(OpenSearchTableHandle table)
         return table.type().equals(OpenSearchTableHandle.Type.QUERY);
     }
 
+    @Override
+    public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjection(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<ConnectorExpression> projections,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!isProjectionPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+        if (containsCallToJsonCastFunction(projections)) {
```

Comment:
without skipping json cast `io.trino.plugin.opensearch.BaseOpenSearchConnectorTest#testQueryTableFunction`, would fail

Reply:
Probably related but I am not sure on that one.

Reply:
What is the actual failure here ?

Reply:
Like this https://github.com/trinodb/trino/actions/runs/9882489305/job/27295482561?pr=22646#step:5:158